pages/resources/[page].js

```diff
@@ -20,35 +31,98 @@ ResourcesPage.propTypes = {
       downvotes: PropTypes.number,
     }),
   ).isRequired,
-  totalPages: PropTypes.number.isRequired,
+  // eslint-disable-next-line react/no-unused-prop-types
+  defaultTotalPages: PropTypes.number.isRequired,
 };
 
-ResourcesPage.getInitialProps = async ({ pathname, query }) => {
-  const { page = 1 } = query;
-
-  const response = await getResourcesPromise({ page });
+ResourcesPage.defaultProps = {
+  currentPage: 1,
+};
 
+ResourcesPage.getInitialProps = async ({ pathname, query }) => {
+  const { q = null, page } = query;
+  /* eventually need some sort of handler to decide which api call to use */
+  const response = !query.search
+    ? await searchResourcesPromise(query)
+    : await getResourcesPromise(query);
+  console.log('configURL=', response.config.url);
+  console.log('QQuery', query);
   const { data: resources, number_of_pages: totalPages, page: currentPage } = response.data;
 
   return {
     currentPage,
     pathname,
     resources,
     totalPages,
+    query,
+    page,
+    q,
   };
 };
 
-function ResourcesPage({ currentPage, pathname, resources, totalPages }) {
+// eslint-disable-next-line react/prop-types
+function ResourcesPage({ currentPage = 1, pathname, resources, totalPages, query }) {
+  // const [categories, setCategories] = useState([]);
+  const [currentResources, setCurrentResources] = useState(resources);
+  const [inputQuery, setInputQuery] = useState(query);
+  const [currentRoute, setCurrentRoute] = useState('');
+
+  // const handlePagination = event => {
+  //   event.preventDefault();
+  //   console.log('slow down there partner');
+  //   const paginationPageNumber = event.currentTarget.getAttribute('value');
+
+  //   router.push({
+  //     pathname: `${pathname.replace('[page]', `${paginationPageNumber}`)}`,
+  //     query: query.q !== null ? { q: query.q } : null,
+  //     shallow: true,
+  //   });
+  // };
+
+  const handleSearch = async searchInputQuery => {
+    setInputQuery(previousState => {
+      return {
+        ...previousState,
+        ...searchInputQuery,
+        search: 'search',
+      };
+    });
+    setCurrentRoute('search');
+
+    const response = await searchResourcesPromise({ ...searchInputQuery, ...query });
+    setCurrentResources(response.data.data);
```

Comment:
couldn't this return tons and tons of resources? Is this query not paginated also?